rules/regexp/headers.lua

```diff
@@ -963,6 +963,26 @@ reconf['OLD_X_MAILER'] = {
   group = 'headers',
 }
 
+-- Detect Apple Mail
+local apple_x_mailer = [[Apple Mail \(((Version )?[1-9]\.[0-9]*|[0-9]*\.[0-9]*\.[0-9]*\.[0-9]*)\)]]
+reconf['APPLE_MAILER'] = {
+  description = 'Sent with Apple Mail',
+  re = string.format('X-Mailer=/^%s/{header}', apple_x_mailer),
+  score = 0.0,
+  group = 'headers',
+}
+
+-- Detect Apple iPhone/iPad Mail
+-- Apple iPhone/iPad Mail X-Mailer contains iOS build number, e. g. 9B206, 16H5, 18G5023c
+-- https://en.wikipedia.org/wiki/IOS_version_history
+local apple_ios_x_mailer = [[i(?:Phone|Pad) Mail \((?:1[AC]|[34][AB]|5[ABCFGH]|7[A-E]|8[ABCEFGHJKL]|9[AB]|\d{2}[A-Z])\d+[a-z]?\)]]
```

Comment:
Can we constrain the last `\d+` to avoid slow performance?

Reply:
It's an existing regex btw, I just moved it to top to be near detect rule

Reply:
I know, but it doesn't mean it that this RE is good :) If we look for a version it is definitely constrained by 1-5 digits (`{1,5}`)

Reply:
its a mix of digits and uppercase letters :)

Reply:
it is to cover this (18G5023c) version for example. If we simplify it - the quality of `FORGED_X_MAILER` will suffer - it is a regular from it: 
https://github.com/rspamd/rspamd/blob/0729c58cb03befdb31b7a1ca56bd052abfd66477/rules/regexp/headers.lua#L1002-L1008

We can leave the complex regex in `FORGED_X_MAILER`, and insert this symbol using a simple regex, only if there no `FORGED_X_MAILER`?

Reply:
I updated regex, please check